### ISSUE TITLE
4장 영화리뷰 fit() 에서 에러가 발생 해결법이 궁금합니다. 

### ISSUE THREAD

**Issue by iamuspace** [ratsgo/nlpbook#40]
안녕하세요. 
책을 잘 보고 있습니다. 

4장 영화리뷰를 실행해 보려고 하는데, 
fit() 실행 부분에서 안되고 있어서
해결 방법이 궁금합니다. 

EPOCH 진행바가 생기는 부분까지 왔는데, 
epoch 0% 에 멈춰있으면서, 
이상하게 .py 파일이 계속 다시 실행되는 것 같습니다. 

아래의 소스와 실행 결과입니다. 

-------------

print('start')

if __name__ == '__main__':
    import torch

    from ratsnlp.nlpbook.classification import ClassificationTrainArguments
    args = ClassificationTrainArguments(
        pretrained_model_name='beomi/kcbert-base',
        downstream_corpus_name='nsmc',
        downstream_model_dir='downstream_model',
        batch_size=32,
        learning_rate=5e-5,
        max_seq_length=128,
        epochs=3,
        tpu_cores=0,
        seed=7
    )

    from ratsnlp import nlpbook

    nlpbook.set_seed(args)
    nlpbook.set_logger(args)

    from Korpora import Korpora

    Korpora.fetch(corpus_name=args.downstream_corpus_name,
                  root_dir=args.downstream_corpus_root_dir,
                  force_download=False)

    from transformers import BertTokenizer

    tokenizer = BertTokenizer.from_pretrained(
        args.pretrained_model_name,
        do_lower_case=False
    )

    from ratsnlp.nlpbook.classification import NsmcCorpus, ClassificationDataset

    corpus = NsmcCorpus()
    train_dataset = ClassificationDataset(
        args=args,
        corpus=corpus,
        tokenizer=tokenizer,
        mode='train'
    )
    print(train_dataset[0])

    from torch.utils.data import DataLoader, RandomSampler

    train_dataloader = DataLoader(
        train_dataset,
        batch_size=args.batch_size,
        sampler=RandomSampler(train_dataset, replacement=False),
        collate_fn=nlpbook.data_collator,
        drop_last=False,
        num_workers=args.cpu_workers,
    )

    from torch.utils.data import SequentialSampler

    val_dataset = ClassificationDataset(
        args=args,
        corpus=corpus,
        tokenizer=tokenizer,
        mode='test'
    )
    val_dataloader = DataLoader(
        val_dataset,
        batch_size=args.batch_size,
        sampler=SequentialSampler(val_dataset),
        collate_fn=nlpbook.data_collator,
        drop_last=False,
        num_workers=args.cpu_workers
    )

    from transformers import BertConfig, BertForSequenceClassification

    pretrained_model_config = BertConfig.from_pretrained(
        args.pretrained_model_name,
        num_labels=corpus.num_labels,
    )
    model = BertForSequenceClassification.from_pretrained(
        args.pretrained_model_name,
        config=pretrained_model_config
    )

    from ratsnlp.nlpbook.classification import ClassificationTask

    task = ClassificationTask(model, args)

    trainer = nlpbook.get_trainer(args)
    trainer.fit(task,train_dataloader,val_dataloader)

----------------------------------------------
실행결과 

start
set seed: 7
INFO:ratsnlp:Training/evaluation parameters ClassificationTrainArguments(pretrained_model_name='beomi/kcbert-base', downstream_task_name='document-classification', downstream_corpus_name='nsmc', downstream_corpus_root_dir='/content/Korpora', downstream_model_dir='downstream_model', max_seq_length=128, save_top_k=1, monitor='min val_loss', seed=7, overwrite_cache=False, force_download=False, test_mode=False, learning_rate=5e-05, epochs=3, batch_size=32, cpu_workers=64, fp16=False, tpu_cores=0)
[Korpora] Corpus `nsmc` is already installed at D:\content\Korpora\nsmc\ratings_train.txt
[Korpora] Corpus `nsmc` is already installed at D:\content\Korpora\nsmc\ratings_test.txt
INFO:ratsnlp:Loading features from cached file /content/Korpora\nsmc\cached_train_BertTokenizer_128_nsmc_document-classification [took 4.358 s]
ClassificationFeatures(input_ids=[2, 2170, 832, 5045, 17, 17, 7992, 29734, 4040, 10720, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], attention_mask=[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], token_type_ids=[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], label=0)
INFO:ratsnlp:Loading features from cached file /content/Korpora\nsmc\cached_test_BertTokenizer_128_nsmc_document-classification [took 1.545 s]
Some weights of the model checkpoint at beomi/kcbert-base were not used when initializing BertForSequenceClassification: ['cls.predictions.transform.LayerNorm.weight', 'cls.seq_relationship.weight', 'cls.predictions.decoder.weight', 'cls.predictions.transform.dense.bias', 'cls.predictions.transform.dense.weight', 'cls.predictions.transform.LayerNorm.bias', 'cls.seq_relationship.bias', 'cls.predictions.bias', 'cls.predictions.decoder.bias']
- This IS expected if you are initializing BertForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing BertForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Some weights of BertForSequenceClassification were not initialized from the model checkpoint at beomi/kcbert-base and are newly initialized: ['classifier.bias', 'classifier.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
Missing logger folder: D:\data-stock\NLP\downstream_model\lightning_logs
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
C:\Anaconda3\envs\trans.gpu2\lib\site-packages\pytorch_lightning\core\optimizer.py:380: RuntimeWarning: Found unsupported keys in the optimizer configuration: {'scheduler'}
  rank_zero_warn(

  | Name  | Type                          | Params
--------------------------------------------------------
0 | model | BertForSequenceClassification | 108 M 
--------------------------------------------------------
108 M     Trainable params
0         Non-trainable params
108 M     Total params
435.680   Total estimated model params size (MB)
Epoch 0:   0%|          | 0/6251 [00:00<?, ?it/s]
start
start
start
start
start
start
start
start
start
start
start

**Comment by iamuspace** [comment]
자체적으로 문제 해결을 하였습니다. 윈도10 사용중인데, 윈도10에서는 멀티스레드 문제가 있어서, args에 명시적으로 일단 
        cpu_workers=0 를 주면 해결되었습니다. 윈도에서 여러 cpu를 사용하려면, 한 줄을 더 써줘야 한다는데 그건 일단 나중으로 미루었습니다.  윈도에서 하시는 분들 참고하세요.
